### ISSUE TITLE
New SymbolProc cop breaks RuboCop when inspecting previously-approved code.

### ISSUE THREAD

**Issue by jdickey** [rubocop/rubocop#1314]
Artefacts are documented in [this Gist](https://gist.github.com/jdickey/9755cddb9b8727870fe6). In summary, 

| Component | Version |
| --- | --- |
| Ruby | ruby 2.1.2p95 (2014-05-08 revision 45877) [x86_64-darwin12.0] |
| rbenv | rbenv 0.4.0-98-g13a474c |
| RuboCop | Commit 05cf121 (or later) |

Commit e9f3428, described as a fix for 05cf121, fails identically.

The description from the [reporting Gist](https://gist.github.com/jdickey/9755cddb9b8727870fe6) is repeated below for reference:

---

We have been using RuboCop for quite some time; one of our [projects](https://github.com/jdickey/new_poc/tree/issue-52v2) tracks the Github repo rather than the released Gem (0.25.0 as I write this).

Commits 05cf121 and e9f3428 (and all later commits) fail when run as a Rake task in otherwise default configuration, or when running the command line

```
rubocop -dnR app/models/cco/blog_cco.rb
```

The debug output from that command is included in this Gist as `output-dump.log`, and the source file demonstrating the failure is also included, as well as the `blo/blog_data_boundary.rb` file which it includes.

RuboCop _does not_ fail when built from Commit 1171580, but works exactly as expected (reporting no offenses in `blog_cco.rb` nor in any other of our current source or spec files).
